File: algorithms/sorting-algorithms/C/sorting_algorithms.c
#include <stdio.h>
#include <stdlib.h> 

void insertion_sort(int array[10], int array_size);
void bubble_sort(int array[10], int array_size);
void print_array(int array[10]);
void swap(int *a, int *b);

void insertion_sort(int array[10], int array_size){
    int i, j, key;
    for(i = 0; i < array_size; ++i){
        key = array[i];
        j = i - 1;
        while(j >= 0 && array[j] > key){
            array[j+1] = array[j];
            j -= 1;
        }
        array[j+1] = key;
    }
    printf("insertion sorted array: ");
    print_array(array);
}

void bubble_sort(int array[10], int array_size){
    int i, j, temp;
    for(i = 0; i < array_size; ++i){
        for(j = 0; j < array_size-i-1; ++j){
            if(array[j] > array[j+1]){
                temp = array[j]; 
                array[j] = array[j+1]; 
                array[j+1] = temp; 
                swap(&array[j], &array[j+1]);
            }
        }
    }
    printf("bubble sorted array: ");
    print_array(array);
}


void print_array(int array[10]){
    int k;
    for(k = 0; k < 10; ++k){
        printf("%d ", array[k]);
    }
}

void swap(int *a, int *b){
    int t = *a;
    *a = *b;
    *b = t;
}

int main(){
    int array[10] = {4, 5, 6, 8, 3, 6, 2, 9, 1, 7};
    int arr_size = sizeof(array) / sizeof(array[0]);
    printf("unsorted array: ");
    print_array(array);
    printf("\n");
    insertion_sort(array, arr_size);
    printf("\n");
    bubble_sort(array, arr_size);
    printf("\narray size: %d\n", arr_size);
    return 0;
}

File: algorithms/sorting-algorithms/python/sorting_algorithms.py
#Sorting Algorithms

def swap(array,i,j):
    array[j], array[i] = array[i], array[j]

def bubble_sort(array):
    for i in range(len(array)):
        for j in range(0, len(array)-i-1):
            if array[j] > array[j+1]:
                swap(array, j, j+1)
    return array

def insertion_sort(array):
    for i in range(1, len(array)):
        key = array[i]
        j = i-1
        while j >= 0 and array[j] > key:
            array[j+1] = array[j]
            j -= 1
        array[j+1] = key
    return array


def merge_sort(array):
    if len(array) > 1:
        m = len(array)//2

        left_array = array[:m]
        right_array = array[m:]
        left_array = merge_sort(left_array)
        right_array = merge_sort(right_array)

        sorted_array = []

        while (len(left_array) > 0) and (len(right_array) > 0):
            if left_array[0] < right_array[0]:
                sorted_array.append(left_array[0])
                left_array.pop(0)
            else:
                sorted_array.append(right_array[0])
                right_array.pop(0)
        
        for i in left_array:
            sorted_array.append(i)
        for i in right_array:
            sorted_array.append(i)
    return array

def partition(array, low, high):
    i = low-1
    
    

def main():
    array = [4, 7, 1, 3, 10, 50, 23, 11, 24]
    print("array: {0}".format(array))
    print("Bubble sorted array: {0}".format(bubble_sort(array)))
    print("Insertion sorted array: {0}".format(insertion_sort(array)))
    print("Merge sorted array: {0}".format(merge_sort(array)))

if __name__ == "__main__":
    main()